File: src/main/java/nz/co/searchwellington/controllers/models/JsonCallbackNameValidator.java
package nz.co.searchwellington.controllers.models;

import org.springframework.stereotype.Component;

@Component
public class JsonCallbackNameValidator {
	
	private static final String VALID_CALLBACK_NAME_REGEX = "[a-z|A-Z|0-9|_]+";
			
	public boolean isValidCallbackName(String callback) {
		return callback.matches(VALID_CALLBACK_NAME_REGEX);
	}
	

}

File: src/main/scala/nz/co/searchwellington/controllers/models/JsonCallbackNameValidator.scala
package nz.co.searchwellington.controllers.models

import org.springframework.stereotype.Component

@Component class JsonCallbackNameValidator {

  private val VALID_CALLBACK_NAME_REGEX: String = "[a-z|A-Z|0-9|_]+"

  def isValidCallbackName(callback: String): Boolean = {
    return callback.matches(VALID_CALLBACK_NAME_REGEX)
  }

}